readme.md

```diff
@@ -923,6 +923,47 @@ const HelloBundle = dynamic({
 export default () => <HelloBundle title="Dynamic Bundle" />
 ```
 
+### Custom `<App>`
+
+<p><details>
+  <summary><b>Examples</b></summary>
+  <ul><li><a href="./examples/layout-component">Using `_app.js` for layout</a></li></ul>
+  <ul><li><a href="./examples/componentdidcatch">Using `_app.js` to override `componentDidCatch`</a></li></ul>
+</details></p>
+
+- Is rendered on both client and server side
+- `this.state` should not be overwritten. Instead create a new component to handle the state ([layout example](./examples/layout-component))
+- Handles calling `getInitialProps` of the **page** `Component`
+- Implements `componentDidCatch` on the client side, and renders the error page accordingly. ([overriding `componentDidCatch` example](./examples/componentdidcatch))
```

Comment:
Let's try not say what've inside `app.js`. But what the user can do with this. Here's a sample:

Next.js has a App layout which initialize pages in your app. You can override it with `pages/_app.js` and implement:

* Layouts works across pages
* Animations in between pages
* Keep the DOM state when navigating page
* Provide additional properties to initial props of pages
* Custom error messages by catching errors
...

Reply:
Updated